client/src/components/ViewContact.jsx

```diff
@@ -2,66 +2,45 @@ import Box from "@mui/material/Box";
 import CircularProgress from "@mui/material/CircularProgress";
 import Snackbar from "@mui/material/Snackbar";
 import Alert from "@mui/material/Alert";
-import { useEffect, useState } from "react";
+import { useEffect, useState, useRef } from "react";
 import EditContactForm from "./EditContactForm";
 import IndividualContact from "./IndividualContact";
 import { fetchContacts } from "../fetchContacts";
 
 const ViewContact = ({ selectedContactId, setContacts }) => {
   const [editButtonClicked, setEditButtonClicked] = useState(false);
-  const [openSnackbar, setOpenSnackbar] = useState(false);
+
   const [snackbarMessage, setSnackbarMessage] = useState("");
+
   const [contactData, setContactData] = useState({
-    newName: "",
-    newEmail: "",
-    newPhone: "",
-    newNotes: "",
-    originalData: {
-      name: "",
-      email: "",
-      phone: "",
-      notes: "",
-    },
+    name: "",
+    email: "",
+    phone: "",
+    notes: "",
   });
+  const originalContactDetails = useRef(null);
 
   useEffect(() => {
     const fetchContact = async () => {
       try {
         const response = await fetch(
           `http://localhost:8080/contacts/${selectedContactId}`
         );
+
         if (!response.ok) {
           throw new Error("Error fetching contact details");
         }
-        const data = await response.json();
-        setContactData((previousData) => ({
-          ...previousData,
-          newName: data.name,
-          newEmail: data.email,
-          newPhone: data.phone,
-          newNotes: data.notes,
-          originalData: {
-            name: data.name,
-            email: data.email,
-            phone: data.phone,
-            notes: data.notes,
-          },
-        }));
+
+        const contactDetails = await response.json();
+
+        setContactData(contactDetails);
+        originalContactDetails.current = contactDetails;
       } catch (error) {
         console.error("Error fetching contact details: ", error);
       }
     };
-    fetchContact();
 
-    if (!selectedContactId) {
-      setContactData((previousData) => ({
-        ...previousData,
-        newName: "",
-        newEmail: "",
-        newPhone: "",
-        newNotes: "",
-      }));
-    }
```

Comment:
No need to clear the state vars here because anytime a contact is selected, the fetch contact details code runs, which overwrites this state.